blockindex/contractstaking/cache.go

```diff
@@ -26,7 +28,10 @@ type (
 		height                uint64                      // current block height, it's put in cache for consistency on merge
 		contractAddress       string                      // contract address for the bucket
 		mutex                 sync.RWMutex                // a RW mutex for the cache to protect concurrent access
+		calculateVoteWeight   calculateVoteWeightFunc     // calculateVoteWeight is a function to calculate vote weight
 	}
+
+	calculateVoteWeightFunc func(v *Bucket, selfStake bool) *big.Int
```

Comment:
since for NFT buckets,  the func is always called with`selfStake = false`, can remove it from the input param?
```
calculateVoteWeightFunc func(v *Bucket) *big.Int
```

Reply:
yes, it's more concise